armada/chartbuilder.py

```diff
@@ -0,0 +1,152 @@
+from hapi.chart.template_pb2 import Template
+from hapi.chart.chart_pb2 import Chart
+from hapi.chart.metadata_pb2 import Metadata
+from hapi.chart.config_pb2 import Config
+from logutil import LOG
+from supermutes.dot import dotify
+import shutil
+import tempfile
+import pygit2
+import os
+import yaml
+
+class ChartBuilder(object):
+    '''
+    This class handles taking chart intentions as a paramter and 
+    turning those into proper protoc helm charts that can be
+    pushed to tiller.
+
+    It also processes chart source declarations, fetching chart
+    source from external resources where necessary
+    '''
+
+    def __init__(self, chart, parent=None):
+
+        # cache for generated protoc chart object
+        self._helm_chart = None
+
+        # record whether this is a dependency based chart
+        self.parent = parent
+
+        # store chart schema
+        self.chart = chart
+
+        # extract, pull, whatever the chart from its source
+        self.source_directory = self.source_clone()
+
+        # cleanup temporary directories
+        self.source_cleanup()
+
+
+    def source_clone(self):
+        '''
+        Clone the charts source
+
+        We only support a git source type right now, which can also
+        handle git:// local paths as well
+        '''
+
+        if self.chart.source.type == 'git':
+
+            tmpdir = tempfile.mkdtemp(prefix='armada', dir='/tmp')
+            self._source_tmp_dir = tmpdir
+            if self.parent:
+                LOG.info("Cloning %s/%s as dependency for %s" % (self.chart.source.location, self.chart.source.subpath, self.parent))
```

Comment:
This is the old way of formatting string. I we should move to the new format `'{} {}'.format('one', 'two')`